src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -83,6 +83,8 @@ public class QueryInsightsSettings {
     public static final String TOP_QUERIES_BASE_URI = PLUGINS_BASE_URI + "/top_queries";
     /** Default prefix for top N queries feature */
     public static final String TOP_N_QUERIES_SETTING_PREFIX = "search.insights.top_queries";
+    /** Default prefix for top N queries grouping feature */
+    public static final String TOP_N_QUERIES_GROUPING_SETTING_PREFIX = "search.insights.top_queries.grouping";
```

Comment:
As I think about it more, I think it still makes sense to combine all the grouping config under the same config group. Let's add a task for this for 2.18 or a future release?